packages/alfa-rules/src/common/predicate/is-rendered.ts

```diff
@@ -27,8 +27,20 @@ export function isRendered(
           return false;
         }
 
+        // <option> elements that are descendants of <select> elements will not
+        // be rendered as only their label is used during rendering.
+        // https://html.spec.whatwg.org/#the-select-element-2
         if (
-          Element.isElement(node) &&
+          hasName("option") &&
+          // <option> elements might be nested within <optgroup> elements so we
+          // need to look at the first two ancestors.
+          node.ancestors().filter(isElement).take(2).some(hasName("select"))
+        ) {
+          return false;
+        }
```

Comment:
🤔 Won't that result in the text node in `<select><option>Foo</option></select>` being seen as not rendered?

Reply:
Yes, which is also intentional. The text within `<option>` elements isn't itself rendered, but is instead used as the label of the corresponding `<select>` element. Stuff like this, for example, has no effect:

```css
option {
  color: red;
}
```

That won't make the text rendered within the `<select>` element red; this will, however:

```css
select {
  color: red;
}
```

Reply:
🤔 but when the select is expanded, the various options are displayed individually, and in that case, the styling on  the option element does apply.
```html
<select style="color: green;">
  <option style="color: blue;">foo</option>
  <option style="color: red;">bar</option>
</select>
```
has its options in various colors when expanded (and its own value in green when collapsed).

OTOH, all this is apparently handled in shadow tree…
Plus, I'm not sure how much we can detect whether the select is expanded or not…

Reply:
That seems UA specific then because this is what I'm seeing:

![image](https://user-images.githubusercontent.com/1174718/102879501-1ffee800-444a-11eb-9400-2df7f606d663.png)

Reply:
The plot thickens... The colours are applied on my end when I make it a multiselect:

![image](https://user-images.githubusercontent.com/1174718/102879639-60f6fc80-444a-11eb-9166-5a7e4623871e.png)

This evidently needs some more refinement 🤔

Reply:
<img width="239" alt="select" src="https://user-images.githubusercontent.com/19969498/102879892-cc40ce80-444a-11eb-80ba-4f3980a2f158.png">
This is what Chrome shows me…

Reply:
> This evidently needs some more refinement 🤔

But in the meantime we might be safe-ish assuming that the select is collapsed since this is likely what Robin gives us currently 🤔